aidb/engine/approx_aggregate_engine.py

```diff
@@ -161,12 +165,12 @@ async def execute_inference_services(self, query: Query, sample_df: pd.DataFrame
           new_query
       )
 
-      input_df = await conn.run_sync(lambda conn: pd.read_sql(text(query_add_filter_key.sql_str), conn))
+      input_df = await conn.run_sync(lambda conn: pd.read_sql_query(text(query_add_filter_key.sql_str), conn))

```

Comment:
What's the difference here?

Reply:
'read_sql' is a convenience wrapper around read_sql_table and read_sql_query.
In PostgreSQL, using read_sql can lead to an sqlalchemy.exc.DBAPIError. I think the driver of PostgreSQL might have compatibility issues with 'read_sql'.

Reply:
Can we move this out to a separate PR? Would be much easier to review